app/services/resource_service.rb

```diff
@@ -15,101 +15,159 @@ def create(resource_class, params)
     # Erlaube nur ein Anlegen von Daten wenn der Nutzer nicht ReadOnly ist.
     raise "Access not permitted" if data_provider.user.read_only_role?
 
+    # cleanup params for given :id
+    @old_resource_id = @params.delete(:id)
+
     # Wenn die Rolle Restricted eine Information anlegt,
     # so ist diese per default nicht sichtbar, es sei denn
     # das Attribute 'visible' wird in den params mitgegeben und ist 'true'
-    if data_provider.user.restricted_role? && resource_class.respond_to?(:visible)
+    if @data_provider.user.restricted_role? && resource_class.respond_to?(:visible)
       @params = { visible: false }.merge(@params)
     end
 
     @resource_class = resource_class
     @resource = resource_class.new(@params)
-    @resource.data_provider = data_provider
+    @resource.data_provider = @data_provider
 
     # skip create if record already exists and new record has the same attribute values as the new
     # record and the record is not marked as 'always_recreate'
-    external_resource = find_external_resource
-    old_record = external_resource.try(:external)
-    if external_resource.present? && unchanged_attributes?(old_record) && !always_recreate?(data_provider, resource_class)
-      external_resource.touch
-      return old_record
+    @external_resource = find_external_resource
+    @old_resource = find_old_resource
+    if @external_resource.present? && unchanged_attributes? && !always_recreate?
+      @external_resource.touch
```

Comment:
Das touch ist nötig, sodass obwohl es keine Veränderung der Daten gab, das `updated_at` erneuert wird. Dies wird nämlich bei Cleanups berücksichtigt und muss somit auf richtigem Stand sein.

Reply:
Kommentar ergänzt in 0215c59